Scripts/BuildPhases/LintAppLocalizedStringsUsage.rb

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env ruby
+
+require 'xcodeproj'
+
+REPO_ROOT = Pathname.new(__dir__) + '../..'
+
+def lint(file_path:, target_name:)
+  violations_count = 0
+  File.foreach(file_path, mode: 'rb:BOM|UTF-8').with_index do |line, line_no|
+    next if line.match? %r(^\s*//) # Skip commented lines
+
+    col_no = line.index('NSLocalizedString')
+    next if col_no.nil?
+
+    puts "#{file_path}:#{line_no+1}:#{col_no+1}: error: Use `AppLocalizedString` instead of `NSLocalizedString` in source files that are used in the `#{target_name}` extension target. See paNNhX-nP-p2 for more info."
+    violations_count += 1
+  end
+  violations_count
+end
+
+## Main ##
+
+project = Xcodeproj::Project.open(REPO_ROOT + 'WordPress/WordPress.xcodeproj')
+targets_to_analyze = if ARGV.count.positive?
+  project.targets.select { |t| t.name == ARGV.first }
+else
+  project.targets.select { |t| t.is_a?(Xcodeproj::Project::Object::PBXNativeTarget) && t.extension_target_type? }
+end
+
+violations_count = 0
+targets_to_analyze.map do |target|
```

Comment:
Nitpick: 

```suggestion
targets_to_analyze.each do |target|
```

Rationale for the record: [`map`](https://apidock.com/ruby/Enumerable/map) "returns a new array with the results of running block once for every element in enum,"  [`each`](https://apidock.com/ruby/Array/each) "calls the given block once for each element in self, passing that element as a parameter."

I formed my understanding of `map` and `each` when dabbling into functional programming at the time Swift was introduced. In that context, it would not be "appropriate" for `map` to perform side effects (the `puts` at line 35) or to change the outer state (updating the `violations_count` value).

Reply:
Ah yeah, that usage of `map` here is a leftover from when I initially tried to write it so that the blocks would return the `violations_count`. But then I realized that using `map` and having the block return each violation count for each target would lead it to not return a single integer value, but an `Array` of ints, one for each target.

Then thought about having the block return a `[target.name, violations_count]` and call `.to_h` on the result to have the whole bock return a Hash of `{ target => violations_count }` with each target. Thought it would be a nice output at first… but then ponder what would be the point of doing that if in the end I don't really care nor use that much level of details since all I needed was to know if there had been at least one violation, to exit with the proper exit code in the end. So making it so I returned a `Hash` felt unnecessary complicated after all.

Then thought about using `reduce` instead of `map`. That would have worked, but would have forced me to make the block take an `|accum, target|`, and then use `accum += violations_count` at the end of the block. Or alternatively, keep the `map` here, and chain it with a call to `.reduce(0, &:+)`. Then again, it all felt less readable in the end… so I ended up with a less "functional-style" but more readable `violations_count = 0` on top and `violations_count +=` inside the loop. In retrospect, I could even have just used `has_violations = false` and `has_violations = true` in the loop, since all I care about is if I had at least one…

Anyway. I'll apply your suggestion here and call it a day 😅